lib/DirectoryWatcher.js

```diff
@@ -235,7 +235,7 @@ DirectoryWatcher.prototype.watch = function watch(filePath, startTime) {
 		if(data) {
 			var ts = data[0] === data[1] ? data[0] + FS_ACCURACY : data[0];
 			if(ts >= startTime)
-				watcher.emit("change", data[1]);
+				watcher.emit("change", filePath, data[1]);
```

Comment:
Not sure, but my first guess is this should depend on `filePath === this.path`:

```suggestion
				if(filePath === this.path)
					watcher.emit("change", data[1]);
				else
					watcher.emit("change", filePath, data[1]);
```

Reply:
arguments of “change" event can it be one ?

https://github.com/webpack/watchpack/blob/6758e8e109e1d1b6d2042397f80f68c7a78b9107/lib/watchpack.js#L98

```
Watchpack.prototype._fileWatcher = function _fileWatcher(file, watcher) {
	watcher.on("change", function(mtime, type) {
		this._onChange(file, mtime, file, type);
	}.bind(this));
	watcher.on("remove", function(type) {
		this._onRemove(file, file, type);
	}.bind(this));
	return watcher;
};

```

Reply:
i will add test.Maybe you are right.